src/core/InputField/InputField.tsx

```diff
@@ -75,7 +78,7 @@ const InputField: React.FC<InputProps> = ({
                 onMouseDown={handleMouseDownPassword}
                 edge="end"
               >
-                {showPassword ? <Visibility /> : <VisibilityOff />}
+                {showPassword ? endIcons[0] : endIcons[1]}
```

Comment:
start and end adornments doesn't need to be an array...it'll be only a single icon...The password field will always have the visibilityIcon and a variable start icon
and other input types will have variable start and end icons...but only one of each...not an array